client.go

```diff
@@ -25,9 +25,62 @@ type Client struct {
 	BaseURL      string
 	StreamingURL string
 	hc           *http.Client
+	token        *oauth2.Token
+}
+
+// WithToken provides an oauth2.Token to the client for auth.
+func WithToken(t *oauth2.Token) func(c *Client) error {
+	return func(c *Client) error {
+		c.token = t
+		return nil
+	}
+}
+
+// WithTokenFile reads a JSON serialized oauth2.Token struct from disk and provides it
+// to the client for auth.
+func WithTokenFile(path string) func(c *Client) error {
+	t, err := loadToken(path)
+	if err != nil {
+		return func(c *Client) error {
+			return err
+		}
+	}
+	return WithToken(t)
+}
+
+// New creates a new Tesla API client. You must provided one of WithToken or WithTokenFile
+// functional options to initialize the client with an OAuth token.
+func New(ctx context.Context, options ...func(*Client) error) (*Client, error) {
```

Comment:
This function returns a client so I wonder if it should be called `NewClient`.  I know that currently conflicts with the old `NewClient` but perhaps we don't need to keep it with this update?

Reply:
Agreed, I was going off the convention of "if there's only one constructor, call it New" but it's not in a `client` package so it's not intuitive `tesla.New` is creating a client. Addressed by just removing the deprecated New* methods.